x/mongo/driver/topology/pool.go

```diff
@@ -176,6 +176,9 @@ func newPool(config poolConfig, connOpts ...ConnectionOption) *pool {
 		conns:                 make(map[uint64]*connection, config.MaxPoolSize),
 		idleConns:             make([]*connection, 0, config.MaxPoolSize),
 	}
+	if pool.maxSize != 0 && pool.minSize > pool.maxSize {
```

Comment:
Can you leave a comment above this line explaining that `minSize` should never exceed `maxSize`?